### PR TITLE
Braze Currents

### PR DESCRIPTION
**Pull request by arnab-p**

### CODE REVIEW

File: v0/sources/message.js [hunk 1]
```diff
@@ -31,6 +30,9 @@ class Message {
   }
 
   setProperty(name, value) {
+    if (!this.properties) {
+      this.properties = {};
```
Comment (prabrishac) [inline v0/sources/message.js:34]: why do we need this?

File: v0/sources/braze/transform.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  if (event.event_type) {
+    const eventType = event.event_type;
+    const message = new Message(`Braze`);
+
+    // since only email status events are supported, event type is always track
+    message.setEventType(messageType);
+
+    // set event name
+    const eventName = eventNameMap[eventType] || eventType;
```
Comment (prabrishac) [inline v0/sources/braze/transform.js:34]: for future support?

File: v0/sources/braze/mapping.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "id": "messageId",
+  "user.user_id": "anonymousId",
+  "user.external_user_id": "userId",
+  "user.device_id": "context.device.id",
+  "user.timezone": "timezone",
```
[the remark this answers is not in the document]
Comment (arnab-p) [reply v0/sources/braze/mapping.json:6]: Fixed. Good catch 👍

File: v0/sources/braze/mapping.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "properties.device_model": "context.device.model",
+  "properties.os_version": "context.os.version",
+  "properties.platform": "context.os.name",
+  "properties.email_address": "traits.email",
+  "properties.user_agent": "context.userAgent",
+  "properties.to_phone_number": "traits.phone",
+  "properties.ad_id": "context.device.advertisingId",
+  "properties.ad_tracking_enabled": "context.device.adTrackingEnabled",
+  "properties": "properties"
```
[the remark this answers is not in the document]
Comment (arnab-p) [reply v0/sources/braze/mapping.json:15]: Modified `properties` handling

File: v0/sources/message.js (hunk 1 above)
Comment (arnab-p) [reply v0/sources/message.js:34]: Not needed. May be late-night coding added this. 🙏 Fixed this.

File: v0/sources/braze/transform.js (hunk 2 above)
Comment (arnab-p) [reply v0/sources/braze/transform.js:34]: Yes

**Review by prabrishac — APPROVED** [review]
Looks good to me 👍

**Review by cpard — APPROVED** [review]
Looks good

**Review by saikumarrs — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**